internal/cmd/editiongen/main.go

```diff
@@ -148,12 +134,8 @@ func main() {
 		log.Fatalf("Usage: %s SCHEMA_REPO ROOT_OUTPUT_DIR", filepath.Base(os.Args[0]))
 	}
 
-	repo, err := git.PlainOpen(os.Args[1])
-	if err != nil {
-		log.Fatalf("%s: %s", filepath.Base(os.Args[0]), err)
-	}
 	for editionName, editionTag := range editionTags {
-		latestEventVersions, err := getLatestEvents(repo, editionTag)
+		latestEventVersions, err := getLatestEvents(context.Background(), os.Args[1], editionTag)
```

Comment:
Perhaps? I picked inside the function since it's the implementation that knows what timeout is reasonable. But yes, callers frequently also have legitimate opinions about the time they want to wait. Either way for me, really.

Reply:
Yeah, just wanted to check that it was a thought through decision.